src/app/inventory/store/override-sockets.ts

```diff
@@ -162,3 +162,21 @@ export function useSocketOverridesForItems(
 
   return [socketOverrides, onPlugClicked, resetSocketOverrides];
 }
+
+/**
+ * Create a socket overrides structure from the item's currently plugged sockets.
+ */
+export function createSocketOverridesFromEquipped(item: DimItem) {
+  const socketOverrides: SocketOverrides = {};
+  for (const socket of item.sockets?.allSockets || []) {
+    // If the socket is plugged and we plug isn't the initial plug we apply the overrides
+    // to the loadout.
+    if (
+      socket.plugged &&
+      socket.plugged.plugDef.hash !== socket.socketDefinition.singleInitialItemHash
```

Comment:
may want to start getting used to `getDefaultPlugHash` instead of `singleInitialItemHash` to avoid bugs with artifice armor

Reply:
Makes sense, I was just moving this from wherever it was before.